official/recommendation/ncf_main.py

```diff
@@ -91,14 +101,18 @@ def _get_ndcg(ranklist, true_item):
       return math.log(2) / math.log(ranklist.index(true_item) + 2)
     return 0
 
+  # Return estimator's last checkpoint as global_step for estimator
+  def _get_global_step(estimator):
+    return int(estimator.latest_checkpoint().split("-")[-1])
```

Comment:
Hmm. This is fragile and unreliable. Can you try estimator.get_variable_value(tf.GraphKeys.GLOBAL_STEP)? (Sorry, that's from memory, so may not be exactly the right syntax.) Or actually reading in the checkpoint and grabbing the global step?